src/main/java/org/motechproject/newebodac/repository/custom/impl/FieldConfigRepositoryImpl.java

```diff
@@ -0,0 +1,53 @@
+package org.motechproject.newebodac.repository.custom.impl;
+
+import java.util.List;
+
+import javax.persistence.EntityManager;
+import javax.persistence.TypedQuery;
+import javax.persistence.criteria.CriteriaBuilder;
+import javax.persistence.criteria.CriteriaQuery;
+import javax.persistence.criteria.ParameterExpression;
+import javax.persistence.criteria.Root;
+
+import org.motechproject.newebodac.domain.BaseEntity;
+import org.motechproject.newebodac.domain.FieldConfig;
+import org.motechproject.newebodac.domain.Vaccinee;
+import org.motechproject.newebodac.domain.enums.EntityType;
+import org.motechproject.newebodac.repository.FieldConfigRepository;
+import org.motechproject.newebodac.repository.custom.FieldConfigRepositoryCustom;
+import org.springframework.beans.factory.annotation.Autowired;
+
+public class FieldConfigRepositoryImpl<E extends BaseEntity> implements FieldConfigRepositoryCustom<E> {
+
+  @Autowired
+  private FieldConfigRepository fieldConfigRepository;
+
+  private final EntityManager em;
+  private final CriteriaBuilder cb;
+
+  public FieldConfigRepositoryImpl(EntityManager em) {
+    this.em = em;
+    this.cb = em.getCriteriaBuilder();
+  }
+
+  /** Checks if value should be unique, then if already exists. */
+
+  @Override
+  public Boolean checkIfUnique(EntityType entityType, BaseEntity entity) {
+    List<FieldConfig> fieldConfigs = fieldConfigRepository.findByEntity(entityType);
+    for (FieldConfig fieldConfig : fieldConfigs) {
```

Comment:
I think you can fetch the configs and move the loop to the fieldConfigService
also you can use the reflections in the service to get the field value
you could throw the exception there and use the field name in it, so it will be more clear for the user where the issue is